Sensor57 TSL2591 Light Sensor Commands

vtHydroponics opened the pull request:
## Description:
Added command functionality to the Sensor57 TSL2591 module via console. The commands allow users to adjust the gain and integration time of the sensor for the light values being returned. The tree is as follows:

`Sensor57 <gain> <integration>`

For `gain`:
- `1` is low
- `2` is medium
- `3` is high
- `4` is max

For `integration` time:
- `1` is 100ms
- `2` is 200ms
- `3` is 300ms
- `4` is 400ms
- `5` is 500ms
- `6` is 600ms

## Checklist:
  - [X] The pull request is done against the latest development branch
  - [X] Only relevant files were touched
  - [X] Only one feature/fix was added per PR and the code change compiles without warnings
  - [X] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [X] The code change is tested and works with Tasmota core ESP32 V.2.0.14
  - [X] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

vtHydroponics committed:
Changes from 11/12/2023. Switches made, need debug

vtHydroponics committed:
Finalized gain/integration adjustment trees

barbudor commented on lib/lib_i2c/Adafruit_TSL2591_Library/Adafruit_TSL2591.cpp:
```diff
@@ -117,7 +117,7 @@ enum
 {
   _initialized = false;
-  _integration = TSL2591_INTEGRATIONTIME_100MS;
+  _integration = TSL2591_INTEGRATIONTIME_300MS;
```
Modifying an external library in that way is a bad idea as it will overwritten if someone upgrade the library

barbudor commented on tasmota/tasmota_xsns_sensor/xsns_57_tsl2591.ino:
```diff
@@ -90,6 +90,58 @@ void Tsl2591Show(bool json)
+    case 1:
+      tsl.setGain(TSL2591_GAIN_LOW);
+      break;
+    case 2:
```
It's probably possible to make a more compact code using an indexed array

vtHydroponics committed:
Fixed the bugs

vtHydroponics commented:
Hello. We fixed the edits you suggested: restored the default Adafruit settings and simplified the multi-line switch-case tree to two matrices and two matrix readers.

vtHydroponics committed:
Outputs in JSON, hex displays as hex was decimal

vtHydroponics commented on tasmota/tasmota_xsns_sensor/xsns_57_tsl2591.ino:
```diff
@@ -90,6 +93,26 @@ void Tsl2591Show(bool json)
+  tsl.setTiming(int_enum_array[value-1]);
+
+  Response_P(PSTR(" {Gain input = %d} {Gain hex = %d}"),XdrvMailbox.payload,tsl.getGain());
+  ResponseAppend_P(PSTR(" {Timing input = %d} {Timing hex = %d}"),value,tsl.getTiming());
```
Fixed, thanks.
